pkg/resource/parameter_group/update_pre_build.go

```diff
@@ -0,0 +1,73 @@
+package parameter_group
+
+import (
+	"context"
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/memorydb"
+)
+
+func (rm *resourceManager) resetParameterGroup(
+	ctx context.Context,
+	desired *resource,
+	latest *resource,
+) (updated *resource, err error) {
+	if (desired.ko.Spec.ParameterNameValues == nil || len(desired.ko.Spec.ParameterNameValues) == 0) && latest.ko.Spec.ParameterGroupName != nil {
+		// Reset all the parameters to default
+		return rm.resetAllParameters(ctx, desired)
+	}
+
+	return nil, nil
+}
+
+// resetAllParameters resets cache parameters for given ParameterGroup in desired custom resource.
+func (rm *resourceManager) resetAllParameters(
+	ctx context.Context,
+	desired *resource,
+) (updated *resource, err error) {
+	input := &svcsdk.ResetParameterGroupInput{}
+	if desired.ko.Spec.ParameterGroupName != nil {
+		input.SetParameterGroupName(*desired.ko.Spec.ParameterGroupName)
+	}
```

Comment:
If we don't set `ParameterGroupName` on the input, then this call would fail anyway. I think we can just assume it will always have a value.